pkg/controller/persistentvolume/persistentvolume_recycler_controller.go

```diff
@@ -40,21 +40,39 @@ var _ volume.VolumeHost = &PersistentVolumeRecycler{}
 // This controller will Recycle those volumes whose reclaim policy is set to PersistentVolumeReclaimRecycle and make them
 // available again for a new claim.
 type PersistentVolumeRecycler struct {
-	volumeController *framework.Controller
-	stopChannel      chan struct{}
-	client           recyclerClient
-	kubeClient       client.Interface
-	pluginMgr        volume.VolumePluginMgr
-	cloud            cloudprovider.Interface
+	volumeController  *framework.Controller
+	stopChannel       chan struct{}
+	client            recyclerClient
+	kubeClient        client.Interface
+	pluginMgr         volume.VolumePluginMgr
+	cloud             cloudprovider.Interface
+	maximumRetryCount int
+	syncPeriod        time.Duration
+	// Local cache of failed recycle / delete operations. Map volume.Name -> status of the volume.
+	// Only PVs in Released state have an entry here.
+	releasedVolumes map[string]releasedVolumeStatus
```

Comment:
I see... as @jsafrane has noted already though we'll pay the price of having scattered state. I wont block this though since you and Jan are aware of the trade off.